### PR TITLE
docs: Improve public docs

### PR DESCRIPTION
**Pull request by jan-auer**
This PR is a rework of the public docs.

- Most of the usage information is now moved to docs from the Readme.
- Missing or badly formatted information has been updated.
- Links to Notion have been removed.
- Sentry-specific information removed, as this is not relevant for public docs.
- Fixed indents for nested lists.
- Added Apple Crash Report endpoint schema

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,20 +33,16 @@ docs: .venv/bin/python
...
+docserver: .venv/bin/python
+	.venv/bin/pip install -U mkdocs
+	.venv/bin/mkdocs serve
+.PHONY: doc
 
-travis-upload-docs: docs install-zeus-cli
+docs-upload: docs
 	cd site && zip -r gh-pages .
 	zeus upload -t "application/zip+docs" site/gh-pages.zip
 .PHONY: travis-upload-docs
```
Comment (untitaker) [inline Makefile]: This target is named the same in Python SDK. If you want it still changed please also change the `.PHONY` line though.

Comment (jan-auer) [reply Makefile]: Right, let's revert this change then.

**Comment by jan-auer** [comment]
@untitaker If you're curious, here's the build: https://zeus.ci/api/repos/gh/getsentry/symbolicator/builds/509/jobs/4/artifacts/e5ed7d06-9f6e-11e9-960c-768b3c1046f8/download

**Comment by jan-auer** [comment]
Docs have been updated: https://getsentry.github.io/symbolicator/

**Pull request merged**